coderd/database/dbmem/dbmem.go

```diff
@@ -3505,6 +3515,18 @@ func (q *FakeQuerier) GetReplicasUpdatedAfter(_ context.Context, updatedAt time.
 	return replicas, nil
 }
 
+func (q *FakeQuerier) GetRuntimeConfig(_ context.Context, key string) (string, error) {
+	q.mutex.Lock()
```

Comment:
Nit: we could use `RLock/RUnlock` here.

Reply:
The fake db uses `Lock` to find any racy stuff with incorrect db usage. Like using a `db` inside a `tx`.

We essentially only support a single "connection" to the db at any given time for `dbmem`.